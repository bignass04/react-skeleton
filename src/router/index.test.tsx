import React from 'react'
import { render, act, screen } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import Router from '.'

vi.mock('mocks/browser', () => ({
  browser: {
    start: vi.fn()
  }
}))

describe('Router', () => {
  test('renders users on the page', async () => {
    let comp
    act(() => {
      comp = render(
        <BrowserRouter>
          <Router />
        </BrowserRouter>
      )
    })
    expect(await screen.findByText('defunkt', undefined, { timeout: 3000 })).toBeInTheDocument()
    expect(await screen.findByAltText('avatar for user defunkt')).toBeInTheDocument()
    expect(comp.container).toMatchSnapshot()
  })
})
