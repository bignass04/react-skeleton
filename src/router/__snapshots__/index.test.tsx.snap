// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Router > renders users on the page 1`] = `
<div>
  <main>
    <h1>
      All Github Users
    </h1>
    <section
      class="users-list"
    >
      <ul>
        <li
          class="list-item home-user-list"
        >
          <a
            class="app-link user-list-name-image"
            href="/user/detail/defunkt"
            target="_self"
          >
            <img
              alt="avatar for user defunkt"
              class="avatar-image"
              src="https://avatars.githubusercontent.com/u/2?v=4"
            />
            <div
              class="user-list-login-name"
            >
              defunkt
            </div>
          </a>
        </li>
        <li
          class="list-item home-user-list"
        >
          <a
            class="app-link user-list-name-image"
            href="/user/detail/ezmobius"
            target="_self"
          >
            <img
              alt="avatar for user ezmobius"
              class="avatar-image"
              src="https://avatars.githubusercontent.com/u/5?v=4"
            />
            <div
              class="user-list-login-name"
            >
              ezmobius
            </div>
          </a>
        </li>
        <li
          class="list-item home-user-list"
        >
          <a
            class="app-link user-list-name-image"
            href="/user/detail/mojombo"
            target="_self"
          >
            <img
              alt="avatar for user mojombo"
              class="avatar-image"
              src="https://avatars.githubusercontent.com/u/1?v=4"
            />
            <div
              class="user-list-login-name"
            >
              mojombo
            </div>
          </a>
        </li>
        <li
          class="list-item home-user-list"
        >
          <a
            class="app-link user-list-name-image"
            href="/user/detail/pjhyett"
            target="_self"
          >
            <img
              alt="avatar for user pjhyett"
              class="avatar-image"
              src="https://avatars.githubusercontent.com/u/3?v=4"
            />
            <div
              class="user-list-login-name"
            >
              pjhyett
            </div>
          </a>
        </li>
        <li
          class="list-item home-user-list"
        >
          <a
            class="app-link user-list-name-image"
            href="/user/detail/wycats"
            target="_self"
          >
            <img
              alt="avatar for user wycats"
              class="avatar-image"
              src="https://avatars.githubusercontent.com/u/4?v=4"
            />
            <div
              class="user-list-login-name"
            >
              wycats
            </div>
          </a>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
